pygmsh/built_in/geometry.py

```diff
@@ -192,6 +192,27 @@ def add_physical_volume(self, volumes, label=None):
         self._add_physical('Volume', volumes, label=label)
         return
 
+    def add_transfinite_lines(self, lines, size):
+        self._GMSH_CODE.append(
+            'Transfinite Line {{{0}}} = {1};'.format(', '.join([l.id for l in lines]), size
+            ))
+
+    def add_transfinite_surface(self, surface, size=None):
+        # a transfinite surface can only have 4 sides
+        assert surface.num_edges == 4
+        if size is not None:
```

Comment:
Just make `size` mandatory?

Reply:
`size` is not mandatory because in general a user can create it's own transfinite lines and then just tell gmsh that the surface is transfinite too. I use it this way, since in general you want to control the transfinite sizes on multiple surfaces that share common edges. The size optional argument is in fact just a shortcut for automating this procedure for simple-enough geometries

Reply:
Alright, fair enough. Perhaps this can be added as an inline comment. (I'll probably forget about this otherwise.)